x-pack/legacy/plugins/lens/public/app_plugin/plugin.tsx

```diff
@@ -94,8 +95,39 @@ export class AppPlugin {
           })
         );
 
+        const redirectTo = (
+          routeProps: RouteComponentProps<{ id?: string }>,
+          addToDashMode: boolean,
+          id?: string
+        ) => {
+          if (!id) {
+            routeProps.history.push('/lens');
+          } else if (!addToDashMode) {
+            routeProps.history.push(`/lens/edit/${id}`);
+          } else if (addToDashMode && id) {
+            routeProps.history.push(`/lens/edit/${id}`);
+            const url = context.core.chrome.navLinks.get('kibana:dashboard');
```

Comment:
This is adding more dependencies on `context`,  which is being deprecated. Is there a better way to do this?